Modify multiple Register methods and configuration files and integrate partial methods, test cases and add separate classes encapsulated by ETCD and ZooKeeper

bingtianyiyan opened the pull request:
// Describe your PR here; eg. Fixes #issueNo

<!--
Thank you for proposing a pull request. This template will guide you through the essential steps necessary for a pull request.
-->
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `dotnet test`.

bingtianyiyan committed:
feat:fix path like "path01\\path02"  special in system

bingtianyiyan committed:
fix:
1.nacos ServerAddresses if is cluster should split ServerList with ,
2.rewrite partion ShenyuNacosRegister.cs method PersistURI and PersistInterface accoding with shenyu nacos client code

bingtianyiyan committed:
feat:
1.add ShenyuConsulRegister.cs init params check
2.edit ShenyuConsulRegister.cs method PersistInterface
3.edit ShenyuConsulRegister.cs method PersistURI

bingtianyiyan committed:
feat:
1.edit ShenyuZookeeperRegister.cs rewrite init zkClient and PersistURI and  PersistInterface
2.zadd ZookeeperUtils

bingtianyiyan committed:
feat:
1.add ZookeeperClientTest  and add apache comment with file
2.add etcdRegister and etcdClientTest

bingtianyiyan committed:
feat:
1. add xx_config.json sample in different RegisterConfigAsync
2. Adds fixed strings for fixed configuration file field for cousul props
3. remove some comment

bingtianyiyan committed:
feat:
1.Path.Combine if secondParms has / will alway return secondParms,so to trim /
2.ectdUtils rent expire change method
3.nacos init add accessKey and secretKey

bingtianyiyan committed:
feat:
1.remove ZookeeperClient same class and unused method
2.add search shengyuAttribute class list

bingtianyiyan committed:
Merge branch 'apache:main' into main

bingtianyiyan committed:
feat:
1.Nacos filed namespace
2.zookeeper modify reconnect timeout

bingtianyiyan committed:
Merge remote-tracking branch 'remotes/origin/dev-20220813'

bingtianyiyan committed:
Merge remote-tracking branch 'remotes/origin/dev-20220813'

bingtianyiyan committed:
feat:
1.remove zookeeper same params

bingtianyiyan committed:
Merge remote-tracking branch 'remotes/origin/dev-20220813'

bingtianyiyan committed:
feat:
1.add ectd docs/etcd_registration.md

bingtianyiyan committed:
feat:
1.zkClient create timeout logical

bingtianyiyan committed:
feat:
1.nacosRegister use ConcurrentQueue replace HashSet
2.registerPath add remove perfix and suffix string to prevent user

bingtianyiyan committed:
Merge remote-tracking branch 'remotes/origin/dev-20220813'

bingtianyiyan committed:
feat:
1.zookeeper_register createMode

bingtianyiyan committed:
feat:
1.edit zookeeper use digest and edit zookeeper Ephemerals node could not create persient child node

bingtianyiyan committed:
feat:
1.zk status change create node

bingtianyiyan committed:
feat：
1.consul config add token

bingtianyiyan committed:
Merge branch 'main' of https://github.com/bingtianyiyan/shenyu-client-dotnet

# Conflicts:
#	client/Apache.ShenYu.AspNetCore/Utils/ShenyuCollectionExtensions.cs
#	client/Apache.ShenYu.Client/Registers/ShenyuEtcdRegister.cs
#	docs/etcd_registration.md
feat: merge main-branch handler error

bingtianyiyan committed:
Merge remote-tracking branch 'remotes/origin/dev-20220813'

bingtianyiyan committed:
feat:
1.edit docs

bingtianyiyan committed:
feat:
1. add apache license header

bingtianyiyan committed:
Merge remote-tracking branch 'remotes/origin/dev-20220813'

hgaol commented:
Can you create an ISSUE and associate with this PR, and describe what problem this PR is trying to solve?

bingtianyiyan committed:
feat:
1.comment out .because now is not have server like zookeeper,etcd started or mock

bingtianyiyan committed:
Merge remote-tracking branch 'remotes/origin/dev-20220813'

bingtianyiyan commented:
> Can you create an ISSUE and associate with this PR, and describe what problem this PR is trying to solve?

ok。
I just want to extract the writing method of the next part. There is no problem with the SDK itself。
